server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -1268,7 +1264,27 @@ public static List<IndexTemplateMetadata> findV1Templates(Metadata metadata, Str
      */
     @Nullable
     public static String findV2Template(Metadata metadata, String indexName, boolean isHidden) {
-        final List<Tuple<String, ComposableIndexTemplate>> candidates = findV2CandidateTemplates(metadata, indexName, isHidden);
+        return findV2Template(metadata, metadata.templatesV2().entrySet(), indexName, isHidden, false);
+    }
+
+    /**
+     * Return the name (id) of the highest matching index template for the given index name. In
+     * the event that no templates are matched, {@code null} is returned.
+     */
+    @Nullable
+    public static String findV2Template(
+        Metadata metadata,
+        Collection<Map.Entry<String, ComposableIndexTemplate>> templateEntries,
```

Comment:
Maybe rename to `candidateTemplates`?

Reply:
That might be confusing with the `candidates` list variable inside the method. I've renamed it to `templates`, which is inline with your other comment.